omx/gstomxaudioenc.c

```diff
@@ -822,21 +865,48 @@ gst_omx_audio_enc_set_format (GstAudioEncoder * encoder, GstAudioInfo * info)
       return FALSE;
   } else {
     if (!(klass->cdata.hacks & GST_OMX_HACK_NO_DISABLE_OUTPORT)) {
-      /* Disable output port */
-      if (gst_omx_port_set_enabled (self->enc_out_port, FALSE) != OMX_ErrorNone)
+      GST_DEBUG_OBJECT (self, "Enabling ports");
+
+      /* Enable input port */
+      if (gst_omx_port_set_enabled (self->enc_in_port, TRUE) != OMX_ErrorNone)
```

Comment:
If this method fails, there is no way to tell in which of this conditionals it was. I recommend adding error logs.

Reply:
The methods called in the conditionals already have error logs internally that indicate what went wrong and also the port index associated to the error. So, adding more error logs could be redundant.